lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -104,16 +141,22 @@ def model_class
       # To be used in the cases where the API does not support list_all with one call.
       #
       def list_in_all_groups
-        array = []
-        mutex = Mutex.new
+        array   = []
+        mutex   = Mutex.new
+        headers = nil
 
         Parallel.each(list_resource_groups, :in_threads => configuration.max_threads) do |rg|
           response = rest_get(build_url(rg.name))
-          results = JSON.parse(response)['value'].map { |hash| model_class.new(hash) }
+          json_response = JSON.parse(response.body)['value']
+          headers = Azure::Armrest::ResponseHeaders.new(response.headers)
+          results = json_response.map { |hash| model_class.new(hash) }
           mutex.synchronize { array << results } unless results.blank?
         end
 
-        array.flatten
+        array = ArmrestCollection.new(array.flatten)
```

Comment:
Nope, good catch. That was something I was tinkering with and forgot to get back to.